src/software/new_geom/line.h

```diff
@@ -19,12 +19,19 @@ class Line final
      */
     explicit Line(const Point& first, const Point& second);
 
+    /**
+     * Returns the coefficient array in the form coeffs[0]*x + coeffs[1]*y + coeffs[2] = 0
+     *
+     * @return the coefficient array
+     */
+    std::array<double, 3> getCoeffs() const;
```

Comment:
Returning an arbitrary 3-long array is a bit ambiguous, but I can see why we're using this here. Could we make it `protected` instead of `public`, then use a test-class that's a `friend` of this class? (I'm not being super detailed here, so feel free to ping me and I can explain exactly what to do in more detail).

This way we can use it for tests, but most users won't have access to it.

Reply:
This was made public so that utility functions like `distance` can access the coefficients. I could wrap those functions in a class and make them static instead, and make it a `friend` class along with the test class. @MathewMacDougall thoughts on this?

Reply:
We could also make `distance` a [friend function](https://www.tutorialspoint.com/cplusplus/cpp_friend_functions.htm) but not sure if that's a road we want to go down. @MathewMacDougall - your thoughts?

Reply:
hmmm yea I'd prefer we don't start making a bunch of `friend functions` since that could get complicated quickly (we might start wanting to do that a lot).

A wrapper `friend` class that makes the functions static also seems complicated.

@garethellis0 what if we returned a "less arbitrary" datatype instead? Maybe a struct with named coefficients or something? Is the main thing you don't like that the numbers in the array have no obvious "meaning", or that it's easy to go out of bounds?

Reply:
I could live with returning a struct here, that was definitely one of the other options I was thinking of.

Reply:
Changed to return a struct